### PR TITLE
Updates BIGNUM and DH models

### PR DESCRIPTION
**Pull request by feliperodri**
*Issue #, if available:*

N/A.

*Description of changes:*

Adds functions and improves conditions in the BIGNUM and DH models. These changes are necessary for the `s2n_dhe` proofs.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: source/dh_override.c [hunk 1]
```diff
@@ -56,7 +70,7 @@ DH *d2i_DHparams(DH **a, const unsigned char **pp, long length) {
 }
 
 int DH_check(DH *dh, int *codes) {
-    assert(openssl_DH_is_valid(dh));
+    assert(dh != NULL);
```
Comment (SaswatPadhi) [inline source/dh_override.c:73]: This part is unclear to me. `DH_size` calls `openssl_DH_is_valid` but here we just check null-ness. Why is that? May be leave a one line comment as to why this change was necessary?

**Review by SaswatPadhi — APPROVED** [review]
LGTM overall, but one small comment:

File: source/dh_override.c (hunk 1 above)
Comment (feliperodri) [reply source/dh_override.c:73]: I created this issue about removing some of the validity functions https://github.com/awslabs/aws-verification-model-for-libcrypto/issues/17
